docs/content/concepts/metadata-tags.mdx

```diff
@@ -40,6 +41,14 @@ How metadata is defined depends on whether you're using assets or ops and jobs:
 
 ---
 
+## How kinds work
+
+Kinds label and categorize definitions in Dagster. Notably, kinds correspond to prominently displayed icons in our visual tools (see XXX for supported visual kinds). An effective proxy for whether something should be a kind is whether or not it is an adjective in your day-to-day language and meaningfully identifies that definition. E.g. If "Is this a dbt asset or a databricks asset?" is a question in your team that would indicate that "dbt" and "spark" are good kinds for your team.
```

Comment:
Regarding this heuristic, I can imagine people using the following sentences:

- "Is this a silver-quality asset or a gold-quality asset?"
- "Is this a raw asset or a staging asset?"
- "Is this a Team A asset or a Team B asset?"

I worry that encouraging users to construct a flat list of:

```python
@asset(kinds=["silver", "raw", "team_a", "s3", "snowflake"])
def foo(): ...
```

is error-prone (what if you use a tool called "silver", but also want to distinguish between "silver quality" and "gold quality", for example).

To reiterate and simplify some comments I had on the original RFC, my overall opinion is that:

- People are going to want / need to categorize their information in a more sophisticated way than a flat list, and we should encourage that (i.e. if you put in the work to map your kinds to categorical information, your searching experience will improve, as will your UI). If we use `kinds`, then there is no convenient way for you to add that information.
- We should not be overally-opinionated in what sort of categorical groupings we allow, and in the long run users should have complete control over the mapping from tags to visual treatment in the UI (i.e. we should allow users to make any grouping they want, not just compute_kind / storage_kind)
- Elevating a flat list of `kinds` to a top-level concept and directly correlating it with UI treatment will make it significantly more difficult to introduce a more flexible concept in the future